pkg/validate/raw/raw.go

```diff
@@ -101,6 +102,7 @@ func Unstructured(objs *objects.Raw) status.MultiError {
 		objects.VisitAllRaw(validate.CRDName),
 		objects.VisitAllRaw(validate.RootSync),
 		objects.VisitAllRaw(validate.RepoSync),
+		objects.VisitAllRaw(validate.SelfReconcile(objs.ReconcilerName)),
```

Comment:
wow, i didn't know we had all this validation! nifty...

do we need an e2e test? 

I don't know where this validation is being called.

Reply:
It is called in the `parseSource` function of both `pkg/parse/root.go` and `pkg/parse/namespace.go`: https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/parse/root.go#L139 and https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/parse/namespace.go#L122.

I added two e2e tests, one for RootSync and one for RepoSync.